CiviGrant - Cleanup managed entities - fixes dev/core#3161

colemanw opened the pull request:
Overview
----------------------------------------
Improves the managed options that come with the CiviGrant extension, to prevent problems documented in [dev/core#3161](https://lab.civicrm.org/dev/core/-/issues/3161).

Technical Details
----------------------------------------
- Ensure labels are translated
- Set stricter cleanup policy for entities that should be removed on uninstall
- Make `grant_status` options unmanaged. They will not be updated by the system ever, but they will be deleted on uninstall because the option group is a managed entity.
- Use the `'match'` option to prevent errors when items already exist (this also allows the 5.47 upgrader to be simplified).

colemanw committed:
CiviGrant - Cleanup managed entities - fixes dev/core#3161

- Ensure labels are translated
- Set stricter cleanup policy for entities that should be removed on uninstall
- Make grant_status options unmanaged. They will not be updated by the system ever,
  but they will be deleted on uninstall because the option group is a managed entity.

totten committed:
Set version to 5.48.1

lcdservices commented:
@colemanw I just ran tests going from < 5.47 to 5.48 and previous modifications to the statuses and types were retained (same results as Tim). I didn't go through an incremental test (upgrading to 5.47 and then later to 5.48). But it looks like Tim has the cause of problems there well documented. If you need me to run through that at some point let me know.

colemanw committed:
CiviGrant - Remove unused managed entities during upgrade

Removes managed entities which may have been added during the 5.47 upgrade,
to prevent the managed system from automatically deleting them.

colemanw commented:
Ok so just to clarify, from your POV @lcdservices this fixes your problem (with sites not yet on 5.47).

@totten I've just pushed up another commit to prevent the managed grant statuses from being deleted. Can you re-test the `474_then_48patch` upgrade?

totten commented:
Merge on pass.

Added forward-port for 5.49-rc: #23179
